### ISSUE TITLE
JavaScript implementation

### ISSUE THREAD

**Issue by jsilvestre** [Tyrox92/DPS_Server#1]
Hey,

I got some work on my spare time. I don't know anything about C++ so I rewrote the server in JavaScript. That way it can be easily hosted somewhere on the internet, and people don't need to tweak their routing configuration.

You can check it here: https://github.com/jsilvestre/gw2-dps-meter-server
It's still basic, but there are some stats available and everyone can see them in the browser (http://host:3000).
Basically, dps variation in % for the group, or individually.

It's a bit clanky because start/end of try uses a simple heuristic.

**Comment by Tyrox92** [comment]
Hey there!

Great work really appreciate it. Finally a programming language i do understand xD
Anyway to give a little more insight into my development. I for myself am a webdeveloper and i am at my limit for C++. I did some Tweaks for the version today but its just a console... even though i formated it right. But i cannot without the help of someone with knowledge get it to show something for everybody.

I personally believe you're on a right track here! But how about we take it a step ahead? Not everybody would have a webhost available, or need to set up one, or install an apache server.
But how about we create a service/online appliaction that let user create rooms.

You could as raidlead or any other partymember goto (http://dps2go.com/) and hit a button saying "create room" or something like that It will then create a room and instance for this player with a random ID and 10 Players can connect to this room with the DPS tool. You could make it all pretty  maybe even sort it after highest damage. and do some fancy stuff there.

If you want to have this as an overlay we could use something like overwatch or similar to make an overlay that displays the HTML page or (dps2go.com/room/1337)


What do you think about this? This would be a final step but this could move the GuildWars 2 Community and Raiding experience to another level.

I am more than willing to help you and create this with you. By any chance some way to contact you? Gw2 Ingame maybe? I am Tyrox.1352 so write me there or answer here so we could talk a little!

**Comment by jsilvestre** [comment]
Hey,

I added you IG so we can chat. I noticed you've been busy with the project, I really like the improvements (https://www.reddit.com/r/Guildwars2/comments/43ka4o/partyraid_dpsmeter/). I must say I think the hosted version seems a better approach to me, because of the routing madness. Also, packaging with node-webkit would allow people without a server to use it. Well, I guess that's another reason to write the server in JavaScript ;-)

As far as the SaaS service is concern, I'm all for it, that's the idea actually. I'm not going to do it by myself, because it's too much work, but I do think it'd be nice, and I'm willing to write the code for it (I just don't want to maintain the server and stuff).

Also, even if the UI is neat in the new improved client, I do think it clutters the game UI without real benefit. What is most interesting to me is the graph at the end that compare multiple tries, which is what I began to do in my code. More importantly, I think it's important to focus on the difference between each try for individuals rather than a ranking between players. Idea is to focus on self-improvements, not shame and blame. Also ranking is pretty inaccurate, for instance in the VG boss fight, I'd rather have 4 people not DPSing and never messing a green area, than wiping because someone forgot or was too late.

Maybe we could have a chat with everyone willing to contribute so we don't slowly pull in 5 directions; but pull x5 in the same. That's what was suggested in the reddit thread, and I'd like to be part of that.
